### DISCUSSION TITLE
Intermittent I2S output and ADC pin input

### DISCUSSION THREAD

**Discussion by al-davis** [pschatzmann/ESP32-A2DP#634] [Q&A] [answered]
I'm having a problem with the I2S output from the ESP32, an intermittend output stream.
I'm using example file bt_music_reciever_32bits exactly as downloaded (with arduino-audio-tools-main library) with an ESP32-WROOM-32 configured in the Arduino IDE as ESP32-WROOM-DA-Module  --  Partition scheme "Minimal SPIFFS(1.9MB APP with OTA/190KB SPIFFS)" ESP32.      Bluetooth is paired with an active audio stream.
Per AudioConfig.h lines 305 - 317  --  Looking at output pins 14 (PIN_I2S_BCK) and 15 (PIN_I2S_WS) I see likely proper clock signals.  Looking at  pin 22  (PIN_I2S_DATA_OUT) I see an intermittent signal (frequent sub second bursts)
By touching pin  GPIO34 (PIN_ADC1)     pin 22 (PIN_I2S_DATA_OUT)  signal becomes steady and good quality audio.  I've looked but don't see reference to this ADC input or understand why it's involved.  What is intended to be the connection here ?

**Comment by pschatzmann** [comment, marked as answer]
From what you describe, I rather think that some of your wires are not connected properly. 
Double check your wiring, try to use some other GND pin or you can [try to use some other i2s pins](https://github.com/pschatzmann/arduino-audio-tools/wiki/External-DAC).

**Comment by al-davis** [comment]
I know, pretty odd !  Also occurs with nothing connected to the ESP32 except the USB cable for power and the scope probe to the data out pin.  I only connected the I2S decoder chip so I could tell if I was actually receiving audio.  I'll try some other pins.

**Reply by pschatzmann** [reply]
I would try another GND pin first: I had some ESP32 where one of the GND pins was not working....

**Reply by al-davis** [reply]
Tried another ground pin, and tried another ESP32 (likely bought at the same time)  -- no help.  Better luck configuring different pins but still a bit flakey (data out comes and goes but at a much lower rate than with the original pin configuration).  I don't have any other BT chips on hand that are compatible with this code.  Maybe an 8266 but that will require some looking at the .h files to get it to compile if that is even compatible.    FWIW  My ESP32 is the same as pictured on this github page

**Reply by pschatzmann** [reply]
I recommend to stick with the regualr 16 bit examples first. 
What I2S module are you using ?
And dont't forget to change the log level to Warning if you have set it to Info...

**Reply by al-davis** [reply]
Logging was not in setup() of bt_music_receiver.ino
adding :
  Serial.begin(115200);
  AudioToolsLogger.begin(Serial, AudioToolsLogLevel::Warning); 
results in the following message soon after pairing :
 152660) I2S: i2s_write(2129): TX mode is not enabled
[E] I2SESP32.h : 104 - size_t audio_tools::I2SDriverESP32::writeBytes(const void*, size_t)
and no signal on data out;

 I2S module is AdaFruit 1334A (PCM5102A chip).

**Reply by pschatzmann** [reply]
Make sure that you use an actual ESP32 core version: the current version is 3.0.7!
Did you follow the [instructions for this module in the Wiki](https://github.com/pschatzmann/arduino-audio-tools/wiki/External-DAC)?

**Reply by al-davis** [reply]
OK, I did have an old version on the IDE.  Upgraded to 3.0.7.   Change the partition scheme to accommodate the increase memory use.
16 bit mode no longer reports the error but no output on data out, both I2S clocks present
32 bit mode no output on data out, both I2S clocks present
The I2S decoder is connected per the WiKi -- signals from the ESP32 are the same whether the I2S decoder is in the circuit or not.

**Reply by al-davis** [reply]
I see this thread is closed.  Have you given up ?
